src/Common/Doctrine/Repository/MetaRepository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Common\Doctrine\Repository;
+
+use Backend\Core\Engine\Exception;
+use Backend\Core\Engine\Model;
+use Common\Uri;
+use SpoonFilter;
+
+class MetaRepository
+{
+    /**
+     * Generate an url, using the predefined callback.
+     *
+     * @param string $URL The base-url to start from.
+     * @param string $class The Fully Qualified Class Name or service name
+     * @param string $method The method that needs to be called
+     * @param array $parameters The parameters for the callback
+     *
+     * @throws Exception When the function does not exist
+     *
+     * @return string
+     */
+    public function generateURL($URL, $class, $method, array $parameters = [])
```

Comment:
I think it will be best to leave it like this for now unless somebody else has a solution